### PR TITLE
Webapp fix

### PR DESCRIPTION
**Pull request by fazelehh**
# Webapp fix

## Summary

Two improvements to the webapp training flow:

**1. Configurable DP-SGD accountant (PRV / RDP)**

The accountant type was hardcoded to `"prv"` everywhere. This PR exposes a PRV/RDP dropdown in the Step 4 DP-SGD config, threads the choice through `TrainParams` on both the backend (`models.py`) and frontend (`api.ts`), and updates the CelebA model handler examples to match.

Also hardens DP-SGD config loading by switching to `.get()` with safe defaults for `epsilon_tolerance` and `eps_error`, preventing `KeyError` on older config files that don't include these fields.

**2. Pretrained image preset (ResNet-18 ImageNet, frozen backbone)**

Adds an "Image (pretrained)" option in Step 3 alongside the existing "Image (scratch)" preset. Uses ResNet-18 with a fully frozen ImageNet backbone — only the final FC layer trains (~160K params vs 11M). Best for small datasets (< 100 samples/class) and DP-SGD runs. Also improves pretrained weight detection to recognise `weights="IMAGENET..."` syntax.

## Changes

- `leakpro/webapp/backend/main.py` — configurable accountant in training loop; new `_PRESET_ARCH_IMAGE_PRETRAINED` architecture; improved pretrained detection
- `leakpro/webapp/backend/models.py` — `accountant: str = "prv"` field on `TrainParams`
- `leakpro/webapp/frontend/src/api.ts` — `accountant?: string` on `TrainParams` interface
- `leakpro/webapp/frontend/src/components/steps/Step3Setup.tsx` — "Image (pretrained)" preset card
- `leakpro/webapp/frontend/src/components/steps/Step4Models.tsx` — accountant dropdown in DP-SGD config
- `examples/mia/celebA_HQ/celebA_model_handler_dpsgd.py` — configurable accountant + safe config defaults
- `examples/mia/celebA_HQ/webapp/celebA_model_handler.py` — same

## How Has This Been Tested?

- Webapp run end-to-end with both PRV and RDP accountant on the CelebA example
- Legacy DP-SGD config (missing `epsilon_tolerance`/`eps_error`) loads without error
- "Image (pretrained)" preset trains successfully on a small dataset with DP-SGD enabled

## Related Pull Requests

- #423 (tabular webapp LoS example)

### CODE REVIEW

**Comment by fazelehh** [comment]
Jira ticket created: [LDG-25](https://aisweden.atlassian.net/browse/LDG-25)

[LDG-25]: https://aisweden.atlassian.net/browse/LDG-25?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: leakpro/webapp/frontend/src/components/steps/Step3Setup.tsx [hunk 1]
```diff
@@ -183,12 +183,25 @@ const PRESETS: Array<{
...
+      optimizer: "Adam (default) or SGD",
+      learning_rate: "0.001",
+      scheduler: "none",
+    },
+  },
+  {
+    id: "image_pretrained",
+    label: "Image (pretrained)",
+    icon: "auto_awesome",
+    desc: "ResNet-18 pretrained on ImageNet — backbone fully frozen, only the final FC layer trains (~160K params). Best for small datasets (< 100 samples/class) and DP-SGD.",
```
Comment (TheColdIce) [inline leakpro/webapp/frontend/src/components/steps/Step3Setup.tsx]: Under DP-SGD this description is inaccurate. The DP path runs ModuleValidator.fix() (main.py:44), which replaces every BatchNorm with a fresh GroupNorm hence discarding the pretrained BN affine + ImageNet running stats, and creating trainable GN params. So with DP-SGD enabled it's not "frozen backbone, only FC trains (~160K params)": it trains FC + all GroupNorm params, and the backbone normalization is no longer the pretrained one. Since this preset is marketed as the DP-SGD choice, the copy should say so. Also "~160K params" assumes ~300 classes (FC = 512Â·num_classes); it's ~5K for 10 classes.

**Review by TheColdIce — CHANGES_REQUESTED** [review]
See inline comment

File: leakpro/webapp/frontend/src/components/steps/Step3Setup.tsx (hunk 1 above)
Comment (fazelehh) [reply leakpro/webapp/frontend/src/components/steps/Step3Setup.tsx]: Good catch — confirmed against the code, you're right on all three points.

Fix (incoming push):
- No longer claims a frozen FC-only backbone under DP-SGD. The Details panel now notes that Opacus's `ModuleValidator.fix()` swaps every BatchNorm for GroupNorm, which drops the pretrained ImageNet normalization stats and makes the GroupNorm affine params trainable — so DP-SGD trains the FC head **plus all GroupNorm params**, not the FC head alone.
- Dropped the misleading "~160K params" — now `512 × num_classes` (e.g. ~5K for 10 classes).
- Same correction applied to the backend arch-template docstring in `main.py`, since that text is written into the user's generated `arch.py`.

**Review by TheColdIce — APPROVED** [review]
Nice!
